include/utils.h

```diff
@@ -1,5 +1,4 @@
-#ifndef UTILS_H
-#define UTILS_H
+#pragma once
 
```

Comment:
give it a big difference between include Guards and pragma once?
i found a heated discussion about it  https://stackoverflow.com/questions/1143936/pragma-once-vs-include-guards but i think both are equal. But if you want we can switch all headers to #pragma once.

Reply:
Haha this was just me desperately trying to figure out what was wrong with my code... I agree they're both probably equal but yea we should just stick to one across this project. We can do that in another PR though